File: 1-request-lifecycle/2-csharp/ItemsService.cs
namespace RequestLifecycle;

/// ItemsService provides in-memory item data so the lesson focuses on the pipeline.
/// </summary>
public class ItemsService
{
    /// List returns the sample item list for the list endpoint.
    /// </summary>
    public IEnumerable<Item> List()
    {
        // Return in-memory data so there is no DB dependency.
        return new[]
        {
            new Item { Id = 1, Name = "keyboard" },
            new Item { Id = 2, Name = "mouse" },
        };
    }

    /// FindById returns the item detail by id that has passed validation.
    /// </summary>
    public Item FindById(int id)
    {
        // Log to prove the action does NOT run when model validation blocks bad input.
        Console.WriteLine($"[service] FindById({id})");
        return new Item { Id = id, Name = $"item-{id}" };
    }
}

File: 1-request-lifecycle/1-java/src/main/java/academy/starci/requestlifecycle/ItemsService.java
package academy.starci.requestlifecycle;

import java.util.List;
import org.springframework.stereotype.Service;

/**
 * ItemsService provides in-memory item data so the lesson focuses on the pipeline.
 */
@Service
public class ItemsService {
    /**
     * findAll returns the sample item list for the list endpoint.
     *
     * @return the in-memory item list
     */
    public List<Item> findAll() {
        // Return in-memory data so there is no DB dependency.
        return List.of(new Item(1, "keyboard"), new Item(2, "mouse"));
    }

    /**
     * findById returns the item detail by id that has passed validation.
     *
     * @param id item id validated by Bean Validation
     * @return the corresponding item
     */
    public Item findById(long id) {
        // Log to prove the handler does NOT run when Bean Validation blocks bad input.
        System.out.println("[service] findById(" + id + ")");
        return new Item(id, "item-" + id);
    }
}
